src/components/stats/StatsGraph.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+
+import d3Graph from '../../utils/d3Graph';
+
+class StatsGraph extends Component {
+  state = {
+    filter: 'all',
+  }
+
+  fetchIfChanged = (_from, _to) => {
+    const { statsFrom, statsTo, fetchStatsByRange } = this.props;
+    if (_from !== statsFrom || _to !== statsTo) {
+      fetchStatsByRange(_from, _to);
+    }
+  }
+
+  getGraphState() {
+    const { filter } = this.state;
+    const { statsByDate } = this.props;
+
+    if (statsByDate && statsByDate.length) {
```

Comment:
This conditional makes me a little uneasy: I can understand `statsByDate` being `[]`, but should the code also allow for its value to be not an Array? In cases like this, it's nice to have React's PropTypes to make things explicit - is something required to be an array and it can be an empty array to indicate 'no data', or is it an optional property - meaning a parent component might not even provide the prop?